File: 897-increasing-order-search-tree/897-increasing-order-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* increasingBST(TreeNode* root) {
        TreeNode* newRoot = new TreeNode(-1);
        TreeNode* temp = newRoot;
        
        solve(root, temp);
        return newRoot -> right;
    }
private:
    // Daily
    void solve(TreeNode* root, TreeNode*& newRoot) {
        if(root == nullptr)
            return;
        solve(root -> left, newRoot);
        
        newRoot -> right = new TreeNode(root -> val);
        newRoot = newRoot -> right;
        
        solve(root -> right, newRoot);
    }
};

File: 897-increasing-order-search-tree/897-increasing-order-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root):
        if root == None:
            return
        
        self.solve(root.left)
        
        self.temp.right = TreeNode(root.val)
        self.temp = self.temp.right
        
        self.solve(root.right)
        
        
    def increasingBST(self, root: TreeNode) -> TreeNode:
        # Daily
        self.newRoot = TreeNode(-1)
        self.temp = self.newRoot
        
        self.solve(root)
        
        return self.newRoot.right